src/ark/segmentation/ez_seg/ez_seg_utils.py

```diff
@@ -0,0 +1,88 @@
+from typing import Generator, Union
+from skimage.io import imread
+from alpineer.image_utils import save_image
+from alpineer import io_utils
+import os
+import shutil
+from tqdm.auto import tqdm
+import numpy as np
+import pathlib
+
+
+def renumber_masks(
+        mask_dir: Union[pathlib.Path, str]
+):
+    """
+    Relabels all masks in mask tiffs so each label is unique across all mask images in entire dataset.
+    Args:
+        mask_dir (Union[pathlib.Path, str]): Directory that points to parent directory of all segmentation masks to be relabeled.
+    """
+    mask_dir = pathlib.Path(mask_dir)
+    io_utils.validate_paths(mask_dir)
+
+    all_images: Generator[pathlib.Path, None, None] = mask_dir.rglob("*.tiff")
+
+    global_unique_labels = 1
+
+    # First pass - get total number of unique masks
+    for image in all_images:
+        img: np.ndarray = imread(image)
+        unique_labels: np.ndarray = np.unique(img)
+        for label in unique_labels:
+            if label != 0:
+                global_unique_labels += 1
```

Comment:
This can be condensed:

```
for image in all_images:
    img: np.ndarray = imread(image)
    unique_labels: np.ndarray = np.unique(img)
    non_zero_labels: np.ndarray = unique_labels[unique_labels != 0]
    global_unique_labels += len(non_zero_labels)
```

avoiding the extra loop of going through each label one by one.

Reply:
Changed & Tested